### PR TITLE
Refactoring the bot.

### PR DESCRIPTION
**Pull request by Xithrius**
Everything can be modular if you try hard enough.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,7 @@
 # holiday-hackathon-bot
 The bot for the end-of-year hackathon &amp; week of workshops organized by the Foothill CS Club.
+
+```shell
+# User "holiday" with the password "placeholder"
+docker run --name holiday -e POSTGRES_PASSWORD=placeholder -d postgres
+```
```
Comment (madhavarshney) [inline README.md]: What's this for?

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -26,4 +26,4 @@ jobs:
 
       - name: Lint with flake8
         run: |
-          flake8 ./bot
+          flake8 .
```
Comment (madhavarshney) [inline .github/workflows/linting.yml]: Why?

File: README.md (hunk 1 above)
Comment (Xithrius) [reply README.md]: So you can set the password for the database, postgres doesn't like when there's no password.

File: bot/__main__.py [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+from bot.bot import HolidayBot
+
+environ["TZ"] = "America/Los_Angeles"
+time.tzset()
+
+intents = Intents.default()
+intents.members = True
+
+bot = HolidayBot(
+    command_prefix="hack ",
```
Comment (madhavarshney) [inline bot/__main__.py]:
```suggestion
    command_prefix=("hack ", "Hack "),
```

File: .github/workflows/linting.yml (hunk 2 above)
Comment (Xithrius) [reply .github/workflows/linting.yml]: Because everything wants to be linted.

File: bot/extensions/activity.py [hunk 4]
```diff
@@ -1,61 +1,60 @@
...
 
 import discord
-from discord.ext import commands
-from PIL import Image, ImageFont, ImageDraw
+from PIL import Image, ImageDraw, ImageFont
+from discord.ext.commands import Cog, Context, command
+
+from bot.bot import HolidayBot
 
+HOST_GUILD = int(environ.get("BOT_HOST_GUILD"))
```
Comment (madhavarshney) [inline bot/extensions/activity.py]: It was fine before imo.

File: Pipfile [hunk 5]
```diff
@@ -29,6 +29,8 @@ python_version = "3.8.6"
 start = "python -m bot"
 lint = "pre-commit run --all-files"
 precommit = "pre-commit install"
+black = "black --check ."
+flake8 = "python -m flake8"
```
Comment (madhavarshney) [inline Pipfile:33]:
```suggestion
flake8 = "python -m flake8 ./bot"
```

File: bot/extensions/warnings.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            msg = f"{e}."
+
+        elif isinstance(e, commands.CheckFailure):
+            msg = "You do not have enough permissions to run this command."
+
+        elif isinstance(e, commands.MissingPermissions):
+            msg = "Bot does not have enough permissions for this command."
+
+        elif isinstance(e, commands.CommandNotFound):
+            msg = "Unknown command."
```
Comment (madhavarshney) [inline bot/extensions/warnings.py]: Bonus: get the prefix from `ctx`

```suggestion
            msg = "Command not found! Use `hack help` to list all available commands."
```

File: postgres/init.sql [hunk 7]
```diff
@@ -1,6 +1,6 @@
 -- CREATE DATABASE holiday_hackathon;
 
-\c holiday_hackathon;
+\c holiday;
```
Comment (madhavarshney) [inline postgres/init.sql]: Leave it at what it was.

File: bot/extensions/warnings.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            msg = f"{e}."
+
+        elif isinstance(e, commands.CheckFailure):
+            msg = "You do not have enough permissions to run this command."
+
+        elif isinstance(e, commands.MissingPermissions):
+            msg = "Bot does not have enough permissions for this command."
+
+        elif isinstance(e, commands.CommandNotFound):
+            msg = f"Command not found! Use `{ctx.prefix}hack help` to list all available commands."
```
Comment (madhavarshney) [inline bot/extensions/warnings.py]:
```suggestion
            msg = f"Command not found! Use `{ctx.prefix}help` to list all available commands."
```

**Pull request merged**